Snappy/SnappyAPI.swift

```diff
@@ -65,103 +65,86 @@ struct SnapchatAPIConstants {
 /// Main struct to SnapchatAPI with uploading/downloading
 struct SnapchatAPI {
     
-    typealias APIResult = Result<AnyObject, NSError>
-    typealias APICompletionHandler = APIResult -> Void
-    typealias APIImageCompletionHandler = Result<UIImage, NSError> -> Void
-    typealias APIMultipartFormData = MultipartFormData -> Void
-    
+    typealias APIResult = Result<Any>
+    typealias APICompletionHandler = (APIResult) -> Void
+    typealias APIImageCompletionHandler = (Result<UIImage>) -> Void
+    typealias APIMultipartFormData = (MultipartFormData) -> Void
     /// Uploads image and will be send to everyone
-    static func upload(image image: UIImage, multipartFormData: APIMultipartFormData? = nil, completion: APICompletionHandler? = nil) {
+    static func upload(image: UIImage, multipartFormData: APIMultipartFormData? = nil,  completion: @escaping APICompletionHandler ) {
         // We transform our image to data that we can send on server.
         // Here we have 80% compression quality, which is 0.8 by default,
         // you can change it by specifying parameter in toData() function.
         // We use guard to be sure that our image can be represented as
         // NSData.
         guard let imageData = image.toData() else { return }
         
+        
         // Using Alamofire we will upload the data on a server and return
         // response with completion block
-        Alamofire.upload(SnapchatAPIConstants.Method.uploadImage,
-            SnapchatAPIConstants.URL.uploadImage,
-            multipartFormData: { multipartData in
-                multipartData.appendBodyPart(data: imageData,
-                    name: SnapchatAPIConstants.Parameters.imageFile,
-                    fileName: "file.jpg",
-                    mimeType: "image/jpeg"
-                )
-                multipartData.appendBodyPart(
-                    data: "\(SnapchatAPIConstants.userId)".dataUsingEncoding(NSUTF8StringEncoding)!,
-                    name: SnapchatAPIConstants.Parameters.fromUserID
-                )
-                multipartFormData?(multipartData)
-            }) { result in
-                switch result {
-                case .Success(let upload, _, _):
-                    upload.responseData { response in
-                        switch response.result {
-                        case .Success(let data):
-                            print("Success: \(String(data: data, encoding: NSUTF8StringEncoding))")
-                        case .Failure(let error):
-                            print("Error: \(error)")
-                        }
+       Alamofire.upload(multipartFormData: { (multipartData) in
```

Comment:
this closure with `multipartData` could be extracted before `Alamofire.upload`, to make it more readable (yeah, this on the left is not readable either :D)